### PR TITLE
DOCS-#2945: Document `modin.experimental.backends.omnisci`

### LINKED ISSUE #2945: Document modin.experimental.backends.omnisci
This should include docstrings and purposes of everything inside `modin.experimental.backends.omnisci` module.

### PR DESCRIPTION
**Pull request by ienkovich**
- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2945 
- [ ] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

**Review by YarShev — CHANGES_REQUESTED** [review]
@ienkovich , a couple of things we need to do.

1) We need to add omnisci query compiler file to pydocstyle check. https://github.com/modin-project/modin/blob/421f68c11a6cc7734fa956362250de0002ff3d2e/.github/workflows/ci.yml#L37

2) We need to add rst documentation for omnisci query compiler as it is done for other ones.

**Comment by ienkovich** [comment]
> 1. We need to add omnisci query compiler file to pydocstyle check.
Done. But I had to modify code to pass the checker.

> 2. We need to add rst documentation for omnisci query compiler as it is done for other ones.
Done

File: docs/flow/modin/experimental/backends/omnisci/query_compiler.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+OmniSci Query Compiler
+""""""""""""""""""""""
+:py:class:`~modin.experimental.backends.omnisci.query_compiler.DFAlgQueryCompiler` implements
```
Comment (YarShev) [inline docs/flow/modin/experimental/backends/omnisci/query_compiler.rst:3]:
```suggestion
""""""""""""""""""""""

:py:class:`~modin.experimental.backends.omnisci.query_compiler.DFAlgQueryCompiler` implements
```

File: docs/flow/modin/experimental/backends/omnisci/index.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+"""""""""""""""
+
+.. toctree::
+    :hidden:
+
+    query_compiler
+
+High-Level Module Overview
+''''''''''''''''''''''''''
+This module contains :py:class:`~modin.experimental.backends.omnisci.query_compiler.DFAlgQueryCompiler`
```
Comment (YarShev) [inline docs/flow/modin/experimental/backends/omnisci/index.rst:13]:
```suggestion
''''''''''''''''''''''''''

This module contains :py:class:`~modin.experimental.backends.omnisci.query_compiler.DFAlgQueryCompiler`
```

File: docs/developer/architecture.rst [hunk 3]
```diff
@@ -317,7 +317,7 @@ by documentation for now, the rest is coming soon...).
    │   ├───experimental
    │   ├── :doc:`experimental </flow/modin/experimental/experimental>`
    │   │   ├───backends
-   │   │   │   └───omnisci
+   │   │   │   └─── :doc:`omnisci </flow/modin/experimental/backends/omnisci/index>`
```
Comment (YarShev) [inline docs/developer/architecture.rst:320]: Please, add a leaf with query compiler rst as it is done for other backends.

File: modin/experimental/backends/omnisci/query_compiler.py [hunk 4]
```diff
@@ -385,26 +388,58 @@ def value_counts(self, **kwargs):
...
         if self._modin_frame._has_unsupported_data:
-            return default_axis_setter(0)(self, index)
-        default_axis_setter(0)(self, index)
-        # NotImplementedError: OmnisciOnRayFrame._set_index is not yet suported
-        # self._modin_frame.index = index
+            default_axis_setter(0)(self, index)
+        else:
+            default_axis_setter(0)(self, index)
+            # NotImplementedError: OmnisciOnRayFrame._set_index is not yet suported
+            # self._modin_frame.index = index
```
Comment (YarShev) [inline modin/experimental/backends/omnisci/query_compiler.py:416]: This change doesn't seem to relate to the issue.

File: modin/experimental/backends/omnisci/query_compiler.py [hunk 5]
```diff
@@ -385,26 +388,58 @@ def value_counts(self, **kwargs):
...
+        ----------
+        columns : list-like
+            New columns.
+        """
         if self._modin_frame._has_unsupported_data:
-            return default_axis_setter(1)(self, columns)
-        self._modin_frame = self._modin_frame._set_columns(columns)
+            default_axis_setter(1)(self, columns)
+        else:
+            self._modin_frame = self._modin_frame._set_columns(columns)
```
Comment (YarShev) [inline modin/experimental/backends/omnisci/query_compiler.py:442]: This change doesn't seem to relate to the issue.

File: modin/experimental/backends/omnisci/query_compiler.py [hunk 6]
```diff
@@ -149,8 +219,6 @@ def getitem_array(self, key):
             )
         return self.getitem_column_array(key)
 
-    # Merge
```
Comment (YarShev) [inline modin/experimental/backends/omnisci/query_compiler.py:152]: This change doesn't seem to relate to the issue.

File: modin/experimental/backends/omnisci/query_compiler.py (hunk 4 above)
Comment (ienkovich) [reply modin/experimental/backends/omnisci/query_compiler.py:416]: This is required to pass doc checker. `return` in the code requires a corresponding section in docstrings.

File: modin/experimental/backends/omnisci/query_compiler.py (hunk 5 above)
Comment (ienkovich) [reply modin/experimental/backends/omnisci/query_compiler.py:442]: This change is also to pass the doc checker.

File: modin/experimental/backends/omnisci/query_compiler.py (hunk 6 above)
Comment (ienkovich) [reply modin/experimental/backends/omnisci/query_compiler.py]: removed

File: docs/developer/architecture.rst (hunk 3 above)
Comment (ienkovich) [reply docs/developer/architecture.rst:320]: fixed

File: docs/flow/modin/experimental/backends/omnisci/index.rst (hunk 2 above)
Comment (ienkovich) [reply docs/flow/modin/experimental/backends/omnisci/index.rst]: fixed

File: docs/flow/modin/experimental/backends/omnisci/query_compiler.rst (hunk 1 above)
Comment (ienkovich) [reply docs/flow/modin/experimental/backends/omnisci/query_compiler.rst]: fixed

File: modin/experimental/backends/omnisci/query_compiler.py (hunk 4 above)
Comment (YarShev) [reply modin/experimental/backends/omnisci/query_compiler.py:426]: ok

File: modin/experimental/backends/omnisci/query_compiler.py (hunk 5 above)
Comment (YarShev) [reply modin/experimental/backends/omnisci/query_compiler.py:452]: ok

**Review by YarShev — APPROVED** [review]
@ienkovich , LGTM, thanks!

**Pull request merged**